src/gossip/service.zig

```diff
@@ -657,116 +665,137 @@ pub const GossipService = struct {
             }
 
             if (pull_requests.items.len > 0) {
-                var x_timer = std.time.Timer.start() catch unreachable;
+                var x_timer = sig.time.Timer.start() catch unreachable;
                 self.handleBatchPullRequest(pull_requests) catch |err| {
                     self.logger.errf("handleBatchPullRequest failed: {}", .{err});
                 };
-                const elapsed = x_timer.read();
-                self.stats.handle_batch_pull_req_time.add(elapsed);
+                const elapsed = x_timer.read().asMillis();
+                self.stats.handle_batch_pull_req_time.observe(@floatFromInt(elapsed));
 
                 pull_requests.clearRetainingCapacity();
             }
 
             if (pull_responses.items.len > 0) {
-                var x_timer = std.time.Timer.start() catch unreachable;
+                var x_timer = sig.time.Timer.start() catch unreachable;
                 self.handleBatchPullResponses(&pull_responses) catch |err| {
                     self.logger.errf("handleBatchPullResponses failed: {}", .{err});
                 };
-                const elapsed = x_timer.read();
-                self.stats.handle_batch_pull_resp_time.add(elapsed);
+                const elapsed = x_timer.read().asMillis();
+                self.stats.handle_batch_pull_resp_time.observe(@floatFromInt(elapsed));
 
                 pull_responses.clearRetainingCapacity();
             }
 
             if (ping_messages.items.len > 0) {
-                var x_timer = std.time.Timer.start() catch unreachable;
+                var x_timer = sig.time.Timer.start() catch unreachable;
                 self.handleBatchPingMessages(&ping_messages) catch |err| {
                     self.logger.errf("handleBatchPingMessages failed: {}", .{err});
                 };
-                const elapsed = x_timer.read();
-                self.stats.handle_batch_ping_time.add(elapsed);
+                const elapsed = x_timer.read().asMillis();
+                self.stats.handle_batch_ping_time.observe(@floatFromInt(elapsed));
 
                 ping_messages.clearRetainingCapacity();
             }
 
             if (pong_messages.items.len > 0) {
-                var x_timer = std.time.Timer.start() catch unreachable;
+                var x_timer = sig.time.Timer.start() catch unreachable;
                 self.handleBatchPongMessages(&pong_messages);
-                const elapsed = x_timer.read();
-                self.stats.handle_batch_pong_time.add(elapsed);
+                const elapsed = x_timer.read().asMillis();
+                self.stats.handle_batch_pong_time.observe(@floatFromInt(elapsed));
 
                 pong_messages.clearRetainingCapacity();
             }
 
             // TRIM gossip-table
-            if (trim_table_timer.read().asMillis() > GOSSIP_TRIM_INTERVAL_MS) {
+            if (trim_table_timer.read().asMillis() > TABLE_TRIM_RATE.asMillis()) {
```

Comment:
Minor preference for using Duration.gt(other) for comparison. The gt method compares the underlying u64 rather than converting both and then comparing

Reply:
mmm not sure whats the best way 

you bring up a good point on the conversion and comparison - i like the '>' operator because its extremely clear what it does - the `gt()` method is easy to understand but theres always the mental energy of reading a new method name and wondering if it does what you expect or something more (ptsd from other codebase methods doing random things)

will change to use `asNanos()` so theres no conversion - hopefully thats ok but curious on your thoughts too

bb22c36

Reply:
Yeah, I agree with your preference for using the operators as it is clearer. I think using asNanos as you suggested is the best way to go